descqa/StellarMassFunction.py

```diff
@@ -0,0 +1,353 @@
+    validation_color = 'black'
+    validation_marker = 'o'
+    default_markers = ['v', 's', 'd', 'H', '^', 'D', 'h', '<', '>', '.']
+    msize = 4 #marker-size
+
+    def __init__(self, z='redshift_true', mass='stellar_mass', Nbins=25, log_Mlo=8., log_Mhi=12.,
+                 observation='', zlo=0., zhi=1.0, zint =0.2, ncolumns=2, **kwargs):
+
+        #catalog quantities
+        self.zlabel = z
+        self.Mlabel = mass
+
+        #z-range  and mass binning
+        self.Nbins = Nbins
+        self.log_Mlo = log_Mlo
+        self.log_Mhi = log_Mhi
+        self.Mbins = np.logspace(log_Mlo, log_Mhi, Nbins+1)
+        self.DM = (log_Mhi - log_Mlo)/Nbins
+
+        #validation data
+        self.validation_data = {}
+        self.observation = observation
+
+        #check for valid observations
+        if not observation:
+            print('Warning: no data file supplied, no observation requested; only catalog data will be shown')
+        elif observation not in self.possible_observations:
+            raise ValueError('Observation {} not available'.format(observation))
+        else:
+            self.validation_data = self.get_validation_data(observation)
+
+        #plotting variables
+        self.ncolumns = int(ncolumns)
+        self._color_iterator = ('C{}'.format(i) for i in count())
+
+        #setup subplot configuration and get z cuts for each plot
+        self.z_lo, self.z_hi = self.init_plots(zlo, zhi, zint)
+        zmin = np.min(self.z_lo)
+        zmax = np.max(self.z_hi)
+        self.filters = [(lambda z: (z > zmin) & (z < zmax), self.zlabel)]
+
+        #setup summary plot
+        self.summary_fig, self.summary_ax = plt.subplots(self.nrows, self.ncolumns, sharex='col')
+        self.summary_fig.text(0.04, 0.5, self.yaxis, va='center', rotation='vertical') #setup a common axis label
+        #could plot summary validation data here if available but would need to evaluate labels, bin values etc.
+        #otherwise setup a check so that validation data is plotted only once on summary plot
+        self.first_pass = True
+
+        self._other_kwargs = kwargs
+
+
+    def init_plots(self, zlo, zhi, zint):
+        #get magnitude cuts based on validation data or default limits (only catalog data plotted)
+        if len(self.validation_data )> 0:
+            z_lo = [self.validation_data[k].get('zlo') for k in self.validation_data.keys() if self.zkey_match in k]
+            z_hi = [self.validation_data[k].get('zhi') for k in self.validation_data.keys() if self.zkey_match in k]
+        else:
+            z_lo = np.arange(zlo, zhi, zint)
+            z_hi = np.arange(zint, zhi+zint, zint)
+
+        #setup number of plots and number of rows required for subplots
+        self.nplots = len(z_lo)
+        self.nrows = (self.nplots+self.ncolumns-1)//self.ncolumns
+
+        #other plotting variables
+        self.markers = iter(self.default_markers)
+        self.yaxis = '$d\phi/d\log M (Mpc^{-3} dex^{-1})$'
```

Comment:
add `r` before the quote symbol `self.yaxis = r'$d\phi/d\log M (Mpc^{-3} dex^{-1})$'`

Reply:
Done